UI/BuildDesignatorWnd.cpp

```diff
@@ -241,25 +244,109 @@ namespace {
             }
         }
         TemporaryPtr<const UniverseObject> source = GetSourceObjectForEmpire(empire_id);
-        return ConditionDescription(location_conditions, GetUniverseObject(candidate_object_id), source);
+
+        if (only_failed_conditions)
+            return ConditionFailedDescription(location_conditions, GetUniverseObject(candidate_object_id), source);
+        else
+            return ConditionDescription(location_conditions, GetUniverseObject(candidate_object_id), source);
     }
 
     boost::shared_ptr<GG::BrowseInfoWnd> ProductionItemRowBrowseWnd(const ProductionQueue::ProductionItem& item,
                                                                     int candidate_object_id, int empire_id)
     {
+        std::string title;
```

Comment:
are these used outside of the if (BuildType) { } blocks? If not, they can be moved into those blocks, and the title one can be made a const reference as well, since it's just set to whatever UserString returns.